code/game/machinery/suit_storage_unit.dm

```diff
@@ -516,7 +549,112 @@
 			span_notice("You escape the cramped confines of [src]!"))
 		open_machine()
 
+/obj/machinery/suit_storage_unit/multitool_act(mob/living/user, obj/item/tool)
+	if(locked)
+		balloon_alert(user, "unlock it first")
+		return TRUE
+
+	access_locked = !access_locked
+	balloon_alert(user, "access panel [access_locked ? "locked" : "unlocked"]")
+	return TRUE
+
 /obj/machinery/suit_storage_unit/attackby(obj/item/I, mob/user, params)
+	if(panel_open && is_operational && istype(I, /obj/item/stock_parts/card_reader))
+		//you need to unlock to perform the operation else anyone can change access on a locked unit
+		if(locked)
+			balloon_alert(user, "unlock first")
+			return
+
+		if(card_reader_installed)
+			balloon_alert(user, "already installed")
+			return
+
+		card_reader_installed = TRUE
+		balloon_alert(user, "card reader installed")
+
+		I.moveToNullspace()
+		qdel(I)
+
+		return TRUE
+
+	if(is_operational && istype(I, /obj/item/modular_computer/pda))
+		//you need to unlock to perform the operation else anyone can change access on a locked closet
+		if(locked)
+			balloon_alert(user, "unlock it first")
+			return TRUE
+
+		if(!card_reader_installed)
+			balloon_alert(user, "requires card reader for task")
```

Comment:
```suggestion
			balloon_alert(user, "no electronics installed!")
```

Reply:
i prefer "needs card reader!". we shouldn't be obscuring that part